pyccel/codegen/printing/fcode.py

```diff
@@ -452,28 +455,151 @@ def _print_Import(self, expr):
         return self._get_statement(code) + '\n'
 
     def _print_PythonPrint(self, expr):
+        end = '\n'
+        sep = ' '
+        code = ''
+        empty_end = FunctionCallArgument(LiteralString(''), 'end')
+        space_end = FunctionCallArgument(LiteralString(' '), 'end')
+        for f in expr.expr:
+            if f.has_keyword:
+                if f.keyword == 'sep':
+                    sep = str(f.value)
+                elif f.keyword == 'end':
+                    end = str(f.value)
+        args_format = []
         args = []
-        n = len(expr.expr)
-        for j, f in enumerate(expr.expr):
+        orig_args = [f for f in expr.expr if not f.has_keyword]
+
+        def formatted_args_to_print(fargs_format, fargs, end):
+
+            def split_long_argument(line):
```

Comment:
Please add docstrings for these functions.

Functions in functions are usually used to handle small functionalities that are only pertinent to the parent function. Once they become to large to glance over they should be independent functions. Here I am struggling to tell which function ends where so I think these functions are causing readability problems

Reply:
Noted, do you think i should make it independent?

Reply:
Yes I think so. I tried to review the function `_print_PythonPrint` and I kept having to skip over ~50 lines (which got worse with my comments on top), this makes it much harder to follow what is happening in the function

Reply:
Noted!